### PR TITLE
Update Liquibase from 3.6.3 to 4.5.0

### PR DESCRIPTION
**Pull request by devurandom**
Release notes: https://docs.liquibase.com/release-notes/home.html

Most important is probably 4.0.0-beta1 [1] which contains two breaking
changes that "will affect anyone using extensions":

> * Extension classes are now loaded via the java.util.ServiceLoader system
>   rather than the custom class finding logic we had before
> * Logging is now based on java.util.Logging with a cleaned up API

The first change does not appear to affect Metabase, but the 2nd likely does.

Hence I followed the log4j JUL adapter guide [2] to bridge
`java.util.logging` (used by Liquibase) to log4j (used by Metabase).
In particular, I switched from log4j-liquibase to log4j-jul and added the
system property `java.util.logging.manager` to the JVM options.

[1]: https://github.com/liquibase/liquibase/releases/tag/v4.0.0-beta1
[2]: https://logging.apache.org/log4j/2.x/log4j-jul/

### CODE REVIEW

**Comment by devurandom** [comment]
```
org.h2.jdbc.JdbcSQLException: Column "VALUES" not found; SQL statement: INSERT INTO "METABASE_FIELDVALUES" ("FIELD_ID", "VALUES", "HUMAN_READABLE_VALUES", "UPDATED_AT", "CREATED_AT") VALUES (2, ?, ?, now(), now()) [42122-197]
```

I saw similar failures when running the tests locally, but assumed I was just missing the database server and things would work when run in CI. I will debug this and notify you, when it actually works.

**Comment by devurandom** [comment]
With my latest push (https://github.com/metabase/metabase/pull/16086#commits-pushed-7d3f889) all test failures have been resolved.

What remains is the database not matching the fixture anymore in three occasions, because of different changeset checksums. If anyone can provide me with the command to update the fixture, that would be nice.

**Comment by devurandom** [comment]
@flamber Do you have a hint for me how to update the fixture? Or is this approach of changing past migrations not something you would accept? If so, what would be a better approach?

**Comment by flamber** [comment]
@devurandom I don't know. You'll have to wait for one of the devs to look at it, but if the changes to migrations doesn't break anything for anyone (current installs, new installs, old install upgrading, etc), then it's probably fine - if it causes problems, then it will likely never be accepted.

**Comment by JonBetts** [comment]
> The new version of Liquibase still doesn't work with the log4j2 adapter, right? See #17399 and [liquibase/liquibase#1777](https://github.com/liquibase/liquibase/issues/1777) / [apache/logging-log4j2#503](https://github.com/apache/logging-log4j2/pull/503)

Yeah it doesn't look like the integration has been fixed yet 😢 I can't upgrade from log4j1 to log4j2 without liquibase logging correctly, so I've got "security concerns" until that happens!

**Comment by camsaul** [comment]
Looks like the upstream issue has been moved to 

https://github.com/liquibase/liquibase/issues/2004

there hasn't been a lot of movement there recently however unfortunately 😢

File: resources/migrations/000_migrations.yaml [hunk 1]
```diff
@@ -288,7 +293,14 @@ databaseChangeLog:
...
           - column:
-              name: rows
+              # liquibase.database.core.H2Database will quote this, because
+              # it is a H2 keyword, which prevents automatic uppercasing by
+              # H2.  So make it uppercase from the start.
+              # An alternative would be to globally set `objectQuotingStrategy`
+              # to `QUOTE_ALL_OBJECTS`, but that will trip the rest of the
+              # Metabase code, because table and column names will then
+              # all be lowercase.
+              name: ROWS
```
Comment (camsaul) [inline resources/migrations/000_migrations.yaml:303]: I think the problem here is that we want this to be `rows` for Postgres and MySQL by `ROWS` for H2

File: bin/start [hunk 2]
```diff
@@ -101,6 +101,7 @@ fi
 JAVA_OPTS="$JAVA_OPTS -XX:+IgnoreUnrecognizedVMOptions" # Don't barf if we see an option we don't understand (e.g. Java 9 option on Java 7/8)
 JAVA_OPTS="$JAVA_OPTS -Djava.awt.headless=true"         # don't try to start AWT. Not sure this does anything but better safe than wasting memory
 JAVA_OPTS="$JAVA_OPTS -Dfile.encoding=UTF-8"            # Use UTF-8
+JAVA_OPTS="$JAVA_OPTS -Djava.util.logging.manager=org.apache.logging.log4j.jul.LogManager" # java.util.logging via log4j: https://logging.apache.org/log4j/2.x/log4j-jul/
```
Comment (camsaul) [inline bin/start:104]: I'm not a huge fan of this since it doesn't work if people just do `java -jar metabase.jar`... can we set this at runtime instead?

**Comment by camsaul** [comment]
Thanks for the PR @devurandom, we ended up upgrading this in #20600

**Pull request closed without merging**